### PR TITLE
Fix #4888 - add RNA delivery report

### LINKED ISSUE #4888: RNA delivery report
**Is your feature request related to a problem in the current program to new available techology or software? Please describe and add links/citations if appropriate.**
When loading e.g. a tomte RNA "case" into a DNA case, the old DNA delivery report would be overwritten. 

**Describe the solution you'd like**
Add a separate RNA_delivery_report case key.

**Describe alternatives you've considered**
Yes, there are now many keys relating specifically to RNA on the cases. It will be nice to refactor this, perhaps as part of the same on the cg automation side. But this is consistent with what it looks like now, and fills a function.

### PR DESCRIPTION
**Pull request by dnil**
This PR adds a functionality or fixes a bug.

Add an RNA delivery report.

<details>
<summary>Testing on cg-vm1 server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. Make sure the PR is pushed and available on [Docker Hub](https://hub.docker.com/repository/docker/clinicalgenomics/scout-server-stage)
1. Fist book your testing time using the Pax software available at [https://pax.scilifelab.se/](https://pax.scilifelab.se). The resource you are going to call dibs on is `scout-stage` and the server is `cg-vm1`.
1. `ssh <USER.NAME>@cg-vm1.scilifelab.se`
1. `sudo -iu hiseq.clinical`
1. `ssh localhost`
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `podman ps`
1. Stop the service with current deployed branch: `systemctl --user stop scout.target`
1. Start the scout service with the branch to test: `systemctl --user start scout@<this_branch>`
1. Make sure the branch is deployed: `systemctl --user status scout.target`
1. After testing is done, repeat procedure at [https://pax.scilifelab.se/](https://pax.scilifelab.se), which will release the allocated resource (`scout-stage`) to be used for testing by other users.
</details>

<details>
<summary>Testing on hasta server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. `ssh <USER.NAME>@hasta.scilifelab.se`
1. Book your testing time using the Pax software. `us; paxa -u <user> -s hasta -r scout-stage`. You can also use the WSGI Pax app available at [https://pax.scilifelab.se/](https://pax.scilifelab.se).
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `conda activate S_scout; pip freeze | grep scout-browser`
1. Deploy the branch to test: `bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_scout -t scout -b <this_branch>`
1. Make sure the branch is deployed: `us; scout --version`
1. After testing is done, repeat the `paxa` procedure, which will release the allocated resource (`scout-stage`) to be used for testing by other users.
</details>

**How to test**:
1. `scout load case -u scout/demo/643594.config.yaml`
2. notice a new RNA report on the case collapsible sidebar
Screenshot 2024-09-25 at 09 02 53

**Expected outcome**:
The functionality should be working
Take a screenshot and attach or copy/paste the output.

**Review:**
- [x] code approved by CR
- [X] tests executed by DN

### CODE REVIEW

File: scout/server/blueprints/cases/templates/cases/collapsible_actionbar.html [hunk 1]
```diff
@@ -144,7 +144,7 @@
     <div href="#" class="bg-dark list-group-item text-white">
       <div class="d-flex flex-row bd-highlight mt-1">
         <div>
-          <span class="menu-collapsed">{{report_key|replace("_"," ")|title}}</span>
+          <span class="menu-collapsed">{{report_key|replace("_"," ")|title|upper_na}}</span>
```
Comment (dnil) [inline scout/server/blueprints/cases/templates/cases/collapsible_actionbar.html:147]: I just couldn't have either "Rna report" or make the key names capital RNA... 😅

File: scout/build/case.py [hunk 2]
```diff
@@ -289,6 +289,7 @@ def build_case(case_data, adapter):
     case_obj["vcf_files"] = case_data.get("vcf_files", {})
     case_obj["omics_files"] = case_data.get("omics_files", {})
     case_obj["delivery_report"] = case_data.get("delivery_report")
+    case_obj["rna_delivery_report"] = case_data.get("rna_delivery_report")
```
Comment (dnil) [inline scout/build/case.py:292]: I have a feeling a lot of the build functionality could be replaced with serialisation aliases and a few functions for the pydantic case model..

Comment (northwestwitch) [reply scout/build/case.py:292]: Mm yes. But another PR

File: scout/server/blueprints/cases/templates/cases/collapsible_actionbar.html (hunk 1 above)
Comment (northwestwitch) [reply scout/server/blueprints/cases/templates/cases/collapsible_actionbar.html:147]: It works fine and it shows:

image

But wound't it be better if we placed it close to the delivery report, or to the gene fusion one?

I would have this order, perhaps

- General report
- mtDNA report
- Delivery report
- RNA Delivery report
- Gene fusion report
- Pipeline version
- Reference info

I know that many cases will lack some of these report, but feels more ordered this way.

**Review by northwestwitch — APPROVED** [review]
Works fine! 💯 

Consider my suggestion to reshuffle the order of the reports on the case side bar.

**Comment by dnil** [comment]
✅ Report loaded ok on stage:
Screenshot 2024-09-25 at 14 07 30

Screenshot 2024-09-25 at 14 04 38

**Comment by northwestwitch** [comment]
> ✅ Report loaded ok on stage: Screenshot 2024-09-25 at 14 07 30
> 
> Screenshot 2024-09-25 at 14 04 38

Nice! 🚀

**Pull request merged**